homeassistant/components/electric_kiwi/__init__.py

```diff
@@ -0,0 +1,71 @@
+"""The Electric Kiwi integration."""
+from __future__ import annotations
+
+import aiohttp
+from electrickiwi_api import ElectricKiwiApi
+from electrickiwi_api.exceptions import AuthException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client, config_entry_oauth2_flow
+
+from . import api
+from .const import DOMAIN
+from .coordinator import ElectricKiwiHOPDataCoordinator
+
+PLATFORMS: list[Platform] = [
+    Platform.SENSOR,
+    Platform.SELECT,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Electric Kiwi from a config entry."""
+    implementation = (
+        await config_entry_oauth2_flow.async_get_config_entry_implementation(
+            hass, entry
+        )
+    )
+
+    session = config_entry_oauth2_flow.OAuth2Session(hass, entry, implementation)
+
+    try:
+        await session.async_ensure_token_valid()
+    except aiohttp.ClientResponseError as err:
+        if 400 <= err.status < 500:
+            raise ConfigEntryAuthFailed(err) from err
+        raise ConfigEntryNotReady from err
+    except aiohttp.ClientError as err:
+        raise ConfigEntryNotReady from err
+
+    ek_api = ElectricKiwiApi(
+        api.AsyncConfigEntryAuth(aiohttp_client.async_get_clientsession(hass), session)
+    )
+    hop_coordinator = ElectricKiwiHOPDataCoordinator(hass, ek_api)
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = {
+        "ek_api": ek_api,
```

Comment:
The coordinator already has the API available, why store it separately again?

Reply:
my lack of time doing OOP and python is showing. Sorted